src/ProductPage/Create.php

```diff
@@ -85,13 +110,17 @@ public function __construct(
 		Page\ContentLoader $contentLoader,
 		Page\ContentEdit $contentEdit,
 		PageType\Collection $pageTypes,
-		PageType\PageTypeInterface $listingPageType,
+		$listingPageType,
 		ProductPageCreateEventDispatcher $dispatcher,
 		ParentPageCreateEventDispatcher $parentDispatcher,
 		Exists $exists,
 		array $productPageTypeMapping
 	)
 	{
+		if ($listingPageType !== null && $listingPageType instanceof PageType\PageTypeInterface) {
```

Comment:
This statement looks wrong, surely it should be `listingPageType !== null && !$listingPageType instanceof PageType\PageTypeInterface`. Need to test on product upload